### DISCUSSION TITLE
Is ESM plugin supported?

### DISCUSSION THREAD

**Discussion by frozenbonito** [serverless/serverless#10629] [Q&A] [answered]
Is ESM plugin supported?
Or are you planning support it?

for example:

```javascript
export default class MyPlugin {
  constructor() {
    // ...
  }
}
```

**Comment by pgrzesik** [comment, marked as answer]
Hey @frozenbonito - at the moment it's not supported and there are no plans at the moment to support it.

**Reply by frozenbonito** [reply]
Thank you for your answer.

Recently, the number of pure ESM packages is increasing.
I think ESM plugin support is needed to make them available for plugin development.

I would like to create a feature request issue (and I'm also ready to create a Pull Request).

**Reply by pgrzesik** [reply]
Hello - feel free to open a PR with that initiative, but I cannot promise that we will be able to dedicate a lot of time for proper consideration at this point in time
